app/worker.py

```diff
@@ -113,6 +114,34 @@ def transcribe(
         "url_path": url_path,
     }
 
+@celery.task(name="detect_language", bind=True)
+def detect_language(self, audio_file_path: str, encode: bool):
+    logger.info(f"Detecting language of {audio_file_path}")
+
+    with open(audio_file_path, "rb") as audio_file:
+        model_name = DEFAULT_MODEL_NAME
+        logger.info(f"Loading model {model_name}")
+        self.update_state(state=STATES["loading_model"], meta={"progress": {"units": "models", "total": 1, "current": 0}})
+        load_model(model_name)
+
+        logger.info(f"Loading audio from {audio_file_path}")
+        self.update_state(state=STATES["encoding"], meta={"progress": {"units": "files", "total": 1, "current": 0}})
+        audio_data = load_audio(audio_file, encode)
+
+        logger.info(f"Detecting audio language")
+        self.update_state(state=STATES["detecting_language"], meta={"progress": {"units": "files", "total": 1, "current": 0}})
+        result = language_detection(audio_data)
+
+    os.remove(audio_file_path)
+
+    logger.info(f"Returning result in job state")
+
+    result_object = { "language_code": result }
+
+    return {
+        "result": result_object
+    }
```

Comment:
noting that the result here is the actual result data, not a reference to a followup url to fetch